_posts/2018-05-18-contributing-to-installer-puppet-modules.md

```diff
@@ -0,0 +1,325 @@
+ostree_repositories : enable-ostree-repo -------------------------------- 0.30s
+katello_repositories : Set up release repositories ---------------------- 0.03s
+foreman_repositories : include_tasks ------------------------------------ 0.03s
+puppet_repositories : Include puppet 5 for RedHat ----------------------- 0.03s
+foreman_installer : include_tasks --------------------------------------- 0.03s
+foreman_installer : include_tasks --------------------------------------- 0.03s
+katello_repositories : include_tasks ------------------------------------ 0.02s
+```
+
+If the command was not successful, please reach out to the [foreman-dev discourse list](https://community.theforeman.org/c/development) (preferred), or #theforeman-dev on freenode. We love having new contributors; someone will help you. It's possible that you did everything correctly and there is a bug in forklift, so don't be afraid to reach out for assistance.
+
+Once you have your centos7-katello-nightly VM running, you are ready to develop!  The next step is to ssh to your instance with `vagrant ssh centos7-katello-nightly`. You should be presented with a shell, and can attempt to reproduce the issue. In our example, we will want to add an option to optionally remove indexes from the `/pub` directory.
+
+The first step is always to reproduce the issue on the nightly VM. If our hypervisor is not our workstation/laptop, we will need to get https and http access to the nightly VM first. To do this we'll set up an SSH tunnel on our hypervisor. 
+
+First, capture the IP address of the guest (*TIP:* You can use `vagrant ssh-config <vm>` to accomplish the same thing):
+
+```
+vagrant@centos7-katello-nightly ~]$ ip addr
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN 
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+    inet6 ::1/128 scope host 
+       valid_lft forever preferred_lft forever
+2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
+    link/ether 52:54:00:6d:3a:d0 brd ff:ff:ff:ff:ff:ff
+    inet 192.168.121.8/24 brd 192.168.121.255 scope global dynamic eth0
+       valid_lft 3420sec preferred_lft 3420sec
+    inet6 fe80::5054:ff:fe6d:3ad0/64 scope link 
+       valid_lft forever preferred_lft forever
+```
+
+There is a lot of output here, but the part we care about is "192.168.121.8".  Next, in a new terminal on your workstation, run this command: `ssh -f -N -L 2000:192.168.121.8:443 user@<hypervisor hostname>`. This will tunnel TCP requests from port 2000 on your workstation through your hypervisor to port 443 on your guest. You can also set up bridged networking, but SSH tunnels are usually quicker to set up. If your VM's IP address changes, you'll need to kill this tunnel and make a new one. If you don't kill the old one first, your new one won't be able to bind to port 2000.
```

Comment:
Any reason you forward here? You can also go to `https://192.168.121.8`. Perhaps you should mention this is needed if you run forklift on another host that localhost.